### PR TITLE
add gemmx visualization scripts

### PR DESCRIPTION
**Pull request by jorendumoulin**
This PR adds the script to generate useful plots for tracking `gemmx` performance.
There are some hardcoded things here and there for now, but it shouldn't be too hard to generalize this to other systems in the future when needed.

As input, this takes the result of `vcd-to-csv` with the scope of `i_snax_streamer_gemmX_streamer_top`.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,3 +4,5 @@ lit
 numpy<2.0.0
 xdsl @ git+https://github.com/xdslproject/xdsl.git@0d7eae3a1c9ee128986d6729ffc1d49786f7819d
 tensorflow-cpu==2.15.0
+pandas
+matplotlib
```
Comment (JosseVanDelm) [inline requirements.txt:8]: can we put this in an optional requirements please?

File: util/visualization/gemmx.py [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+            x + dx, y + dy, operand, color=color, fontsize=10, ha="center", va="center"
+        )
+
+    ax.set_aspect("auto")
+    plt.tight_layout()
+
+    # Save the plot
+    output_file = os.path.join(output_path, "banking_conflicts.pdf")
+    plt.savefig(output_file)
+    plt.close()
```
Comment (JosseVanDelm) [inline util/visualization/gemmx.py:253]: Maybe put this in `main()`?

File: util/visualization/gemmx.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    ax.set_xticks(indices + bar_width / 2)
+    ax.set_xticklabels(banks, rotation=90)
+    ax.legend()
+
+    plt.tight_layout()
+
+    # Save the plot
+    output_file = os.path.join(output_path, "nb_of_stalls_per_bank.png")
+    plt.savefig(output_file)
+    plt.close()
```
Comment (JosseVanDelm) [inline util/visualization/gemmx.py:140]: maybe put this in `main()`?

**Review by JosseVanDelm — COMMENTED** [review]
Some additional comment sprinkling

File: util/visualization/gemmx.py (hunk 3 above)
Comment (jorendumoulin) [reply util/visualization/gemmx.py:140]: no because I may want to use the functions separately

File: requirements.txt (hunk 1 above)
Comment (jorendumoulin) [reply requirements.txt]: done!

File: util/visualization/gemmx.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+
+def plot_banking_conflicts(df, output_path):
+    """
+    Creates a plot showing bank fires and stalls over time.
+    """
+    # num_cycles = 140
+    # start_cycle = 3650
```
[the remark this answers is not in the document]
Comment (jorendumoulin) [reply util/visualization/gemmx.py]: ok

**Review by JosseVanDelm — APPROVED** [review]
Thank you :pray: can you also update the README with some words about the different optional dependencies please? Thanks again!

**Pull request merged**